### PR TITLE
Feat input component

### PR DESCRIPTION
**Pull request by hubdrpl**

### CODE REVIEW

File: src/components/Button.js [hunk 1]
```diff
@@ -202,7 +202,7 @@ class Button extends React.PureComponent {
             .drui-button--disabled,
             .drui-button--disabled:hover {
               background: ${darkTheme.button.primary.backgroundColorDisabled};
-              border: 1px solid ${darkTheme.button.primary.borderColor};
+              border: 1px solid ${darkTheme.button.secondary.borderColor};
```
Comment (rafalswietek) [inline src/components/Button.js]: Why did you change that? It's a style declaration for the primary button, not secondary. Was the color wrong?

File: src/themes/DarkTheme.js [hunk 2]
```diff
@@ -32,6 +32,31 @@ export default {
     backgroundColor: colors.shark,
   },
 
+  inputWindow: {
```
Comment (rafalswietek) [inline src/themes/DarkTheme.js]: Please, keep in themes files only variables that we use to style components.
If we want to use some of those colors to style the story decorator, use those colors directly from the `colors` object.

File: src/components/Input.js [hunk 3]
```diff
@@ -0,0 +1,230 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import defaultTheme from '../themes/DefaultTheme';
+import darkTheme from '../themes/DarkTheme';
+import { ErrorIcon } from './icons';
+
+class Input extends React.PureComponent {
+  constructor(props) {
```
Comment (rafalswietek) [inline src/components/Input.js]: Unnecessary constructor

File: src/components/Input.js [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+      readOnly,
+      disabled,
+      onFocus,
+      onKeyPress,
+      onChange,
+    } = this.props;
+
+    return (
+      <div className="drui-inputContainer">
+        {label && <span className="drui-input-label">{label}</span>}
```
Comment (rafalswietek) [inline src/components/Input.js]: Please, stick with the class naming as we use in our other projects.
E.g. this class name should be `drui-input__label` or `drui-inputLabel`.
Please, amend the rest of those class names too ;)

File: src/components/Input.js [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+            autoFocus={autoFocus}
+            readOnly={readOnly}
+            disabled={disabled}
+            onFocus={onFocus}
+            onKeyPress={onKeyPress}
+            onChange={onChange}
+          />
+          {error && <ErrorIcon className="drui-input-error__errorIcon" />}
+        </div>
+        {info && !error && <div className="drui-input-info"><span>{info}</span></div>}
```
Comment (rafalswietek) [inline src/components/Input.js]: I'm not sure if the info element should disappear when some error occurs :/

Comment (hubdrpl) [reply src/components/Input.js]: Im not sure either, this is how it looks in dashboard now.

File: src/components/Button.js (hunk 1 above)
Comment (hubdrpl) [reply src/components/Button.js]: It was wrong

File: .storybook/config.js [hunk 6]
```diff
@@ -14,7 +14,7 @@ const ThemesDecorator = (storyFn) => (
         .storyWrapperLight {
           width: 400px;
           padding: 40px;
-          background: ${defaultTheme.window.backgroundColor};
```
Comment (rafalswietek) [inline .storybook/config.js]: If we're not using `defaultTheme` and `darkTheme` here anymore we can remove imports as well

File: stories/Input.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import Input from '../src/components/Input';
+import darkTheme from '../src/themes/DarkTheme';
```
Comment (rafalswietek) [inline stories/Input.js]: Unnecessary import

File: src/components/Input.js (hunk 5 above)
Comment (rafalswietek) [reply src/components/Input.js]: Please, let the info element to be displayed no matter if there is an error, or not ;)

**Comment by rafalswietek** [comment]
These are the things we need to fix too:
 * wrong story naming convention in the last input story
 * missing transition effect (let's have a nice 0.2s ease transitions)
 * missing purple glow in active inputs
 * wrong input height (szould be 40px, not 44px)
 * missing eye icon button to show / hide password in the password field
 * wrong label and info weight and color (weight should be 400, and the color should be #5E646E)
 * wrong spacing between label, sublabel, info and input (should be 6px instead of 10px)
 * wrong error message position (should be right under the input, not under the info and the spacing between an error and an input should be 0px)

File: src/components/Input.js [hunk 8]
```diff
@@ -49,11 +68,18 @@ class Input extends React.PureComponent {
             onFocus={onFocus}
             onKeyPress={onKeyPress}
             onChange={onChange}
+            ref={(input) => { this.input = input; }}
           />
-          {error && <ErrorIcon className="drui-inputError__errorIcon" />}
+          {initialType === 'password' && !error && (
```
Comment (rafalswietek) [inline src/components/Input.js]: Ok, 2 things here.
First of all, lets make this component accept a prop called e.g. `passwordVisibilityToggle` with a boolean value that will indicate if we should show that password visibility toggle button or not (ofc only when `this.props.type === 'password'`. It will also make `this.initialType` useless.
Also, we can't hide that button when an error occurs, because this will be probably the moment when the users are going to need it the most ;)

File: src/components/Input.js [hunk 9]
```diff
@@ -49,11 +68,18 @@ class Input extends React.PureComponent {
...
+              type === 'text'
+                ? <PublicIcon className="drui-input__icon drui-input__passwordVisibilityIcon" onClick={this.togglePasswordVisibility} />
```
Comment (rafalswietek) [inline src/components/Input.js]: Please, move props declarations to new lines, so we won't have such a long single lines od code

File: src/components/Input.js [hunk 10]
```diff
@@ -49,11 +68,18 @@ class Input extends React.PureComponent {
...
+                : <PrivateIcon className="drui-input__icon drui-input__passwordVisibilityIcon" onClick={this.togglePasswordVisibility} />
```
Comment (rafalswietek) [inline src/components/Input.js]: Same here ;)

File: src/components/Input.js [hunk 11]
```diff
@@ -3,14 +3,29 @@ import PropTypes from 'prop-types';
...
 import darkTheme from '../themes/DarkTheme';
-import { ErrorIcon } from './icons';
+import { PublicIcon, PrivateIcon, ErrorIcon } from './icons';
 
 class Input extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      type: props.type,
```
Comment (rafalswietek) [inline src/components/Input.js]: I would suggest to change that state property to `passwordVisible: props.passwordVisible` and use it to indicate if the password should be visible or not by default. This will let us to create password inputs where the password will be visible by default ;)
